corehq/messaging/smsbackends/telerivet/views.py

```diff
@@ -115,132 +124,118 @@ def page_context(self):
             'gateway_list_url': reverse(DomainSmsGatewayListView.urlname, args=[self.domain]),
         }
 
-    @property
-    def unexpected_error(self):
-        return _(
-            "An unexpected error occurred. Please start over and if the "
-            "problem persists, please report an issue."
-        )
 
-    @allow_remote_invocation
-    def get_last_inbound_sms(self, data):
-        request_token = data.get('request_token')
-        if not request_token:
-            return {'success': False}
-
-        webhook_secret = self.get_cached_webhook_secret(request_token)
-        if not webhook_secret:
-            return {'success': False}
-
-        result = IncomingRequest.get_last_sms_by_webhook_secret(webhook_secret)
-        if result:
-            return {
-                'success': True,
-                'found': True,
-            }
-        else:
-            return {
-                'success': True,
-                'found': False,
-            }
-
-    def trim(self, value):
-        if isinstance(value, six.string_types):
-            return value.strip()
-
-        return value
-
-    @allow_remote_invocation
-    def send_sample_sms(self, data):
-        request_token = data.get('request_token')
-        if not self.get_cached_webhook_secret(request_token):
-            return {
-                'success': False,
-                'unexpected_error': self.unexpected_error,
-            }
-
-        outgoing_sms_form = TelerivetOutgoingSMSForm({
-            'api_key': data.get('api_key'),
-            'project_id': data.get('project_id'),
-            'phone_id': data.get('phone_id'),
-        })
+@require_GET
```

Comment:
@orangejenny it looks like these three new views aren't using any auth. the request_token is providing some level of protection, but it would be good to have the same level of auth as with the main `TelerivetSetupView`

Reply:
Good point, I looked at `TelerivetSetupView` but not at the superclasses. https://github.com/dimagi/commcare-hq/pull/21698 should address this.